src/test/java/org/HdrHistogram/HistogramTest.java

```diff
@@ -394,21 +394,76 @@ public void testSubtract() throws Exception {
         Assert.assertEquals(2L, histogram.getCountAtValue(testValueLevel));
         Assert.assertEquals(2L, histogram.getCountAtValue(testValueLevel * 1000));
         Assert.assertEquals(4L, histogram.getTotalCount());
+
+        verifyMaxValue(histogram);
+        verifyMaxValue(other);
+    }
+
+    @Test
+    public void testSubtractToZeroCounts() {
+        Histogram histogram = new Histogram(highestTrackableValue, numberOfSignificantValueDigits);
+        histogram.recordValue(testValueLevel);
+        histogram.recordValue(testValueLevel * 1000);
+        Assert.assertEquals(1L, histogram.getCountAtValue(testValueLevel));
+        Assert.assertEquals(1L, histogram.getCountAtValue(testValueLevel * 1000));
+        Assert.assertEquals(2L, histogram.getTotalCount());
+
         // Subtracting down to zero counts should work:
         histogram.subtract(histogram);
         Assert.assertEquals(0L, histogram.getCountAtValue(testValueLevel));
         Assert.assertEquals(0L, histogram.getCountAtValue(testValueLevel * 1000));
         Assert.assertEquals(0L, histogram.getTotalCount());
-        // But subtracting down to negative counts should not:
-        boolean thrown = false;
+
+        verifyMaxValue(histogram);
+    }
+
+    @Test
+    public void testSubtractToNegativeCountsThrows() {
+        Histogram histogram = new Histogram(highestTrackableValue, numberOfSignificantValueDigits);
+        Histogram other = new Histogram(highestTrackableValue, numberOfSignificantValueDigits);
+        histogram.recordValue(testValueLevel);
+        histogram.recordValue(testValueLevel * 1000);
+        other.recordValueWithCount(testValueLevel, 2);
+        other.recordValueWithCount(testValueLevel * 1000, 2);
+
         try {
-            // This should throw:
             histogram.subtract(other);
+            fail();
         } catch (IllegalArgumentException e) {
```

Comment:
With the new subtraction logic, this was throwing AIOOBE instead because it (correctly) assessed that `histogram`'s new max (0) was too small for `other`. So, I split up the tests so that I didn't have to rearrange this one too vigorously to avoid that.